TACC/Core-Portal
# Task/WP-729: Mutation hook: Move files

## Overview

We are working on updating our current mutations to utilize TypeScript and React Query instead of React Redux. This should improve the speed and functionality of all portal systems going forward.

## Related

* [WP-729](https://tacc-main.atlassian.net/browse/WP-729)

## Changes

I converted `useMove.js` to `useMove.ts` and created a mutation that moves one or more files to a new location by updating its path.

## Testing

1. Navigate to a file that you want to move.
2. Select it and click on Move.
3. Select the folder that you want to move the selected file into and click on Move.
4. You should get a notification stating that the file is moved to its new location.

## UI

![image](https://github.com/user-attachments/assets/1a5e06fa-1653-4640-ad8d-c2b4bb1ccc0f)

## Notes
